patches/tModLoader/Terraria/Utils.TML.cs

```diff
@@ -275,4 +275,50 @@ private static void AddArgToDictionary(string text, ref string text2, ref Dictio
 
 		text2 = "";
 	}
+
+	/// <summary>
+	/// Creates a <see cref="Rectangle"/> from the provided corners. They do not need to be in a specific order.
+	/// </summary>
+	public static Rectangle CornerRectangle(Point pointA, Point pointB)
+	{
+		int left = Math.Min(pointA.X, pointB.X);
+		int top = Math.Min(pointA.Y, pointB.Y);
+		int width = Math.Abs(pointA.X - pointB.X);
+		int height = Math.Abs(pointA.Y - pointB.Y);
+		return new Rectangle(left, top, width, height);
+	}
+
+	/// <inheritdoc cref="CornerRectangle(Point, Point)"/>
+	public static Rectangle CornerRectangle(Vector2 pointA, Vector2 pointB) => CornerRectangle(pointA.ToPoint(), pointB.ToPoint());
+
+	/// <summary>
+	/// Creates a <see cref="Rectangle"/> containing all of the provided points.
+	/// </summary>
+	public static Rectangle BoundingRectangle(params Point[] points)
+	{
+		if (points.Length == 0)
+			return new Rectangle();
+		var rectangle = new Rectangle(points[0].X, points[0].Y, 0, 0);
+		foreach (var point in points)
+			rectangle = rectangle.Including(point);
+		return rectangle;
+	}
+
+	/// <inheritdoc cref="BoundingRectangle(Point[])"/>
+	public static Rectangle BoundingRectangle(params Vector2[] vectors) => BoundingRectangle(vectors.Select(ToPoint).ToArray());
```

Comment:
Since we don't have a value linq system imported, I recommend duplicating the method above